module-system/sov-state/src/storage.rs

```diff
@@ -237,14 +237,20 @@ pub trait Storage: Clone {
     /// Opens a storage access proof and validates it against a state root.
     /// It returns a result with the opened leaf (key, value) pair in case of success.
     fn open_proof(
-        &self,
         state_root: Self::Root,
         proof: StorageProof<Self::Proof>,
     ) -> Result<(StorageKey, Option<StorageValue>), anyhow::Error>;
 
     /// Indicates if storage is empty or not.
     /// Useful during initialization.
     fn is_empty(&self) -> bool;
+
+    /// Returns the value corresponding to the key or None if key is absent and a proof to
+    /// get the value.
+    /// Only supposed to be used in native context
```

Comment:
We can't assume this here as  the `Storage` trait has to work with any context. We have this: 
https://github.com/Sovereign-Labs/sovereign-sdk/blob/nightly/module-system/sov-state/src/storage.rs#L273